client-src/elements/utils.ts

```diff
@@ -586,61 +576,75 @@ interface FeatureUpdateInfo {
 
 // Prepare feature/stage changes to be submitted.
 export function formatFeatureChanges(
-  fieldValues,
-  featureId,
+  fieldValues: FieldInfo[],
+  featureId: number,
   formStageId?: number
 ): UpdateSubmitBody {
   let hasChanges = false;
-  const featureChanges = {id: featureId};
-  // Multiple stages can be mutated, so this object is a stage of stages.
-  const stages = {};
-  // When editing an individual stage, always provide stage ID so that
-  // active_stage_id can be set by the server.
-  if (formStageId) {
-    stages[formStageId] = {id: formStageId};
-  }
-  for (const {name, touched, value, stageId, implicitValue} of fieldValues) {
-    // Only submit changes for touched fields or accuracy verification updates.
+  const featureChanges: FeatureUpdateInfo = {id: featureId};
+
+  const directFeatureChanges: { name: string; value: any }[] = [];
+  const pendingStageChanges: {
+    stageId: number;
+    originalName: string;
+    value: any;
+  }[] = [];
+
+  for (const field of fieldValues) {
+    const {name, touched, value, stageId, implicitValue} = field;
     if (!touched) {
       continue;
     }
+    hasChanges = true;
 
-    // Arrays should be submitted as comma-separated strings.
-    let formattedValue = value;
-    if (Array.isArray(formattedValue)) {
-      formattedValue = formattedValue.join(',');
-    }
+    const formattedValue = Array.isArray(value) ? value.join(',') : value;
```

Comment:
These off-topic changes make it harder to see the actual upgrade that was requested.